crates/ruff_linter/resources/test/fixtures/flake8_pyi/PYI019_0.py

```diff
@@ -135,3 +135,18 @@ class NoReturnAnnotations:
     def m[S](self: S, other: S): ...
     @classmethod
     def n[S](cls: type[S], other: S): ...
+
+class MethodsWithBody:
+    def m[S](self: S, other: S) -> S:
+        x: S = other
+        return x
+
+    @classmethod
+    def n[S](cls: type[S], other: S) -> S:
+        x: type[S] = type(other)
+        return x()
+
+class StringizedReferencesAreTooComplicated:
+    def m[S](self: S) -> S:
+        x = cast("S", self)
+        return x
```

Comment:
Could you add a test where `S` gets rebound (or deleted) inside the method body? Just to make sure that these references don't get rewritten.